core/state_processor.go

```diff
@@ -197,8 +202,14 @@ func ProcessBeaconBlockRoot(beaconRoot common.Hash, vmenv *vm.EVM, statedb *stat
 		To:        &params.BeaconRootsAddress,
 		Data:      beaconRoot[:],
 	}
-	vmenv.Reset(NewEVMTxContext(msg), statedb)
-	statedb.AddAddressToAccessList(params.BeaconRootsAddress)
+	txctx := NewEVMTxContext(msg)
+	vmenv.Reset(txctx, statedb)
+	if vmenv.ChainConfig().Rules(num, true, time).IsEIP2929 {
```

Comment:
Shouldn't it be `IsEIP4762`?

Reply:
nope, access lists are a concept of 2929 that is deactivated with 4762

Reply:
Is there any particular reason why you are placing all these calls to `AccessList` within `isEIP2929` clauses? 
I mean, you could just let the call happen, couldn't you?

Reply:
I could, but it's not supposed to happen (even if it has no obvious impact afaict)